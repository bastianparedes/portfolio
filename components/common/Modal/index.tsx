import React from 'react';
import styles from './styles.module.scss';
import Shadow from '../Shadow';
import { MdClose } from 'react-icons/md';

interface props {
  children?: React.ReactNode;
  setModalVisible: (boolean: boolean) => void;
}

const Modal = ({ children, setModalVisible }: props): JSX.Element => {
  const handleOnClose = (): void => {
    setModalVisible(false);
  };

  return (
    <Shadow setVisible={setModalVisible}>
      <div className={styles.modal}>
        <button
          data-testid="data-testid-button-closer-modal"
          className={styles.button}
          onClick={handleOnClose}
        >
          <MdClose />
        </button>
        <div className={styles.childrenContainer}>{children}</div>
      </div>
    </Shadow>
  );
};

export default Modal;
