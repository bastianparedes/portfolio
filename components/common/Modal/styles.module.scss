@import 'styles/_breakpoints';

$background-color: rgb(255, 255, 255);

.modal {
  min-width: 200px;
  min-height: 100px;
  position: absolute;
  left: 50vw;
  top: 50vh;
  background-color: $background-color;
  transform: translate(-50%, -50%);
  border-radius: 5px;
}

.button {
  display: flex;
  align-content: center;
  justify-content: center;
  position: absolute;
  right: 0;
  transform: translate(50%, -50%);
  padding: 0;
  background-color: $background-color;
  border: 3px solid;
  border-radius: 50%;
  font-size: 40px;
  color: rgb(0, 0, 0);
}

.childrenContainer {
  overflow: auto;
}

@include phone {
  .modal {
    width: 100%;
    min-height: 10%;
    max-height: 90%;
    position: absolute;
    top: auto;
    bottom: 0;
    left: 0;
    transform: none;
    border-radius: 10px 10px 0 0;
  }

  .button {
    border: none;
    transform: none;
    margin: 10px;
  }
}
