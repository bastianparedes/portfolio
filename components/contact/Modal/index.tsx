import React from 'react';
import styles from './styles.module.scss';

interface props {
  success: boolean;
  text: string;
  setModalVisible: (boolean: boolean) => void;
}

const Modal = ({ success, text, setModalVisible }: props): JSX.Element => {
  const svg = success ? 'Ã‰xito' : 'Fracaso';

  const handleOnClose = (): void => {
    setModalVisible(false);
  };

  return (
    <div className={styles.modal}>
      {svg}
      <span>{text}</span>
      <button
        data-testid="data-testid-button-closer-modal"
        onClick={handleOnClose}
      ></button>
    </div>
  );
};

export default Modal;
