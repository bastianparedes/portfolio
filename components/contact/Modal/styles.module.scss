.main {
  min-width: 100vw;
  min-height: calc(100vh - 3rem);
  background-color: lightblue;
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
}

.title {
  text-align: center;
}
