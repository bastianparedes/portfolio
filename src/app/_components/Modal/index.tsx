'use client';

import React from 'react';
import { IoMdClose } from 'react-icons/io';

interface Props {
  children: React.ReactNode;
  closeModal: () => void;
}

const Component = ({ children, closeModal }: Props) => {
  return (
    <div
      className="h-svh w-svw bg-gray-800/80 fixed top-0 left-0 flex justify-center items-center z-10"
      onClick={closeModal}
    >
      <div
        className="absolute bottom-0 md:relative"
        onClick={(e) => e.stopPropagation()}
      >
        <button
          onClick={closeModal}
          className="text-4xl aspect-square justify-center items-center left-0 top-0 bg-green-400 rounded-full -translate-x-1/2 -translate-y-1/2 z-10 absolute hidden md:flex"
        >
          <IoMdClose />
        </button>
        <div className="relative bg-slate-900 w-screen max-h-[90svh] rounded-tl-3xl rounded-tr-3xl overflow-y-auto p-10 md:rounded-3xl md:rounded-tl-none md:max-w-[80svw] md:w-screen">
          {children}
        </div>
      </div>
    </div>
  );
};

export default Component;
