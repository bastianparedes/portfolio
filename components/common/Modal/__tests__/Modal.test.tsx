import React from 'react';
import Modal from '..';
import { fireEvent, render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

jest.mock(
  '../../Shadow',
  () =>
    ({ children }: { children: React.ReactNode }) =>
      <>{children}</>
);

describe('<Modal />', () => {
  it('should render', () => {
    const props = {
      setModalVisible: jest.fn()
    };

    const { container } = render(<Modal {...props} />);
    expect(container).toMatchSnapshot();
  });

  it('should close modal when button is clicked', () => {
    const props = {
      success: true,
      text: 'Success',
      setModalVisible: jest.fn()
    };

    const { getByTestId } = render(<Modal {...props} />);
    const button = getByTestId('data-testid-button-closer-modal');
    act(() => {
      fireEvent.click(button);
    });

    expect(props.setModalVisible).toBeCalledTimes(1);
    expect(props.setModalVisible).toBeCalledWith(false);
  });
});
