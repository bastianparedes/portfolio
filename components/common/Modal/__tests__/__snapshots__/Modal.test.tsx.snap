// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should render 1`] = `
<div>
  <div
    class="modal"
  >
    <button
      class="button"
      data-testid="data-testid-button-closer-modal"
    />
    <div
      class="childrenContainer"
    />
  </div>
</div>
`;
